jcl/src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -283,16 +283,19 @@ private static int countDuplicates(StackTraceElement[] currentStack, StackTraceE
  * @return an array of StackTraceElement representing the stack
  */
 StackTraceElement[] getInternalStackTrace() {
-
 	if (disableWritableStackTrace) {
 		return	ZeroStackTraceElementArray;
 	}
 
-	if (stackTrace == null) {
-		stackTrace = J9VMInternals.getStackTrace(this, true);
+	StackTraceElement[] localstackTrace = stackTrace;
+	if (localstackTrace == null) {
+		// Assign the result to a local variable to avoid race condition
+		// that stackTrace might be set back to null.
```

Comment:
```suggestion
		// Assign the result to a local variable to avoid refetching
		// the instance variable and any memory ordering issues
```

Reply:
I think we should reword the comment as I still don't see a race condition.  I'm not convinced my phrasing is better but without knowing what the race is, I'd rather avoid suggesting there is one

Reply:
@DanHeidinga Thanks, agreed that this might be a memory ordering issue instead. Updated the comments accordingly.
Please start the PR tests.

Reply:
Also changed `localstackTrace` to `localStackTrace`.